Automattic/wp-calypso
# Store Orders: Add "Delete" button to orders

This PR adds the functionality to delete an order off a site. This includes the data-layer handler, actions, and a reducer to track the "isDeleting" state. It also adds a "Delete" button to a single order, which pops up an "Are you sure" dialog before deleting the order. Fixes #15680

New option in order header:

![buttons](https://user-images.githubusercontent.com/541093/35711795-c1e1e850-078c-11e8-9e43-dedab91fc924.png)

After clicking Delete, this dialog makes sure they really mean it:

![screen shot 2018-02-01 at 8 17 36 pm](https://user-images.githubusercontent.com/541093/35711827-f86547fa-078c-11e8-8ab1-23f43b075dce.png)

A few questions:

- ❓  Technically this trashes the order, not "delete permanently". I think this might be for the best, in case the store owner is really not thinking and needs to recover the order, but they can't really do that via Calypso (so either they'd dive into wp-admin, or contact support).
- ❓  I'd like to customize the _Are you sure_ text a little, like "you can refund by X, or cancel by Y" but I'm not sure if/how that can fit. Copy help appreciated 🙂 

**To test**

- Open any order from `/store/orders/:siteId`
- Click "Delete"
- Click "OK"
- It should give you a notice, then redirect to the orders list, and once it finishes reloading it should not have that order in the list
- Try to delete another order, and cancel out, it should _not_ delete the order
